Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -158,6 +158,20 @@ IPCCommandResult ES::GetTitleID(const IOCtlVRequest& request)
   return GetDefaultReply(IPC_SUCCESS);
 }
 
+static bool UpdateUIDAndGID(const IOS::ES::TMDReader& tmd)
+{
+  IOS::ES::UIDSys uid_sys{Common::FromWhichRoot::FROM_SESSION_ROOT};
+  const u32 uid = uid_sys.GetOrInsertUIDForTitle(tmd.GetTitleId());
```

Comment:
Sure, I've done this.
Btw, as you may have noticed, I chose to make `UpdateUIDAndGID` take a TMD, because there are no cases where we will want to update the PPC's UID and GID with arbitrary values.